classes/navigationMenu/NavigationMenuItemDAO.inc.php

```diff
@@ -0,0 +1,531 @@
+<?php
+
+/**
+ * @file classes/navigationMenu/NavigationMenuItemDAO.inc.php
+ *
+ * Copyright (c) 2014-2017 Simon Fraser University
+ * Copyright (c) 2000-2017 John Willinsky
+ * Distributed under the GNU GPL v2. For full terms see the file docs/COPYING.
+ *
+ * @class NavigationMenuItemDAO
+ * @ingroup navigationMenuItem
+ * @see NavigationMenuItem
+ *
+ * @brief Operations for retrieving and modifying NavigationMenuItem objects. NMI = NavigationMenuItem
+ */
+
+import('lib.pkp.classes.navigationMenu.NavigationMenu');
+import('lib.pkp.classes.navigationMenu.NavigationMenuItem');
+
+class NavigationMenuItemDAO extends DAO {
+	/**
+	 * Constructor
+	 */
+	function __construct() {
+		parent::__construct();
+	}
+
+	/**
+	 * Retrieve a navigation menu item by ID.
+	 * @param $navigationMenuItemId int
+	 * @return NavigationMenuItem
+	 */
+	function getById($navigationMenuItemId) {
+		$params = array((int) $navigationMenuItemId);
+		$result = $this->retrieve(
+			'SELECT	* FROM navigation_menu_items WHERE navigation_menu_item_id = ?',
+			$params
+		);
+
+		$returner = null;
+		if ($result->RecordCount() != 0) {
+			$returner = $this->_fromRow($result->GetRowAssoc(false));
+		}
+		$result->Close();
+		return $returner;
+	}
+
+	/**
+	 * Retrieve a navigation menu item by path.
+	 * @param $contextId int Context Id
+	 * @param $path string
+	 * @return NavigationMenuItem
+	 */
+	function getByPath($contextId, $path) {
+		$params = array($path, (int) $contextId);
+		$result = $this->retrieve(
+			'SELECT	* FROM navigation_menu_items WHERE path = ? and context_id = ?',
+			$params
```

Comment:
Generally speaking, it's a little less verbose to just put `array($path, (int) $contextId)` here in cases where you don't need to dynamically construct the `$params` array. (Just stylistic.)